scripts/test-util.js

```diff
@@ -0,0 +1,297 @@
+                + 'com.hazelcast.remotecontroller.Main --use-simple-server> rc_stdout.txt 2>rc_stderr.txt', [], {
+                stdio: ['ignore', 'inherit', 'inherit'],
+                shell: true
+            });
+        }
+
+    } else {
+        if (background) {
+            rcProcess = spawn(`nohup java -Dhazelcast.enterprise.license.key=${HAZELCAST_ENTERPRISE_KEY} `
+                + `-cp ${CLASSPATH} com.hazelcast.remotecontroller.Main --use-simple-server> rc_stdout.txt`
+                + '2>rc_stderr.txt &', [], {
+                stdio: ['ignore', 'inherit', 'inherit'],
+                shell: true
+            });
+        } else {
+            rcProcess = spawn(`java -Dhazelcast.enterprise.license.key=${HAZELCAST_ENTERPRISE_KEY} `
+                + `-cp ${CLASSPATH} com.hazelcast.remotecontroller.Main --use-simple-server > rc_stdout.log `
+                + '2> rc_stderr.log', [], {
+                stdio: ['ignore', 'inherit', 'inherit'],
+                shell: true
+            });
+        }
+    }
+
+    console.log('Please wait for Hazelcast Remote Controller to start ...');
+
+    const retryCount = 10;
+
+    for (let i = 0; i < retryCount; i++) {
+        console.log('Trying to connect to 127.0.0.1:9701...');
+        const addressReachable = await isAddressReachable('127.0.0.1', 9701, 5000);
+        if (addressReachable) {
+            return;
+        }
+        await new Promise(r => setTimeout(r, 1000));
+    }
+    throw new Error(`Could not reach to 127.0.0.1:9701 after trying ${retryCount} times.`);
+}
+
+if (process.argv.length === 3) {
+    if (process.argv[2] === 'unit') {
+        MOCHA_COMMAND = 'mocha "test/unit/**/*.js"';
+        testType = 'unit';
+    } else if (process.argv[2] === 'integration') {
+        MOCHA_COMMAND = 'mocha "test/integration/**/*.js"';
+        testType = 'integration';
+    } else if (process.argv[2] === 'alltests') {
+        MOCHA_COMMAND = 'mocha "test/**/*.js"';
+        testType = 'alltests';
+    } else if (process.argv[2] === 'startrc') {
+        downloadDependencies();
+        startRC(true).then(() => {
+            console.log('Hazelcast Remote Controller is started!');
+            process.exit(0);
+        }).catch(err => {
+            console.log('Could not start Hazelcast Remote Controller due to an error:');
+            throw err;
+        });
+    } else {
+        throw 'Operation type can be one of "unit", "integration", "alltests", "startrc"';
+    }
+} else {
+    throw 'Usage: node <script-file> <operation-type>. '
+    + 'Operation type can be one of "unit", "integration", "alltests", "startrc"';
+}
+
+if (!fs.existsSync('./lib')) {
+    console.log('lib folder does not exists compiling..');
+    spawnSync('npm run compile', [], {
+        stdio: 'inherit',
+        shell: true
+    });
+}
+
+if (testType === 'unit') {
+    console.log(`Running unit tests... Mocha command: ${MOCHA_COMMAND}`);
+    spawnSync(MOCHA_COMMAND, [], {
```

Comment:
When run outside of npm script, this (and other mocha related) command fails
```bash
$ node scripts/test-util.js unit
Running unit tests... Mocha command: mocha "test/unit/**/*.js"
/bin/sh: 1: mocha: not found
```

The simplest fix would be to change the command to `node node_modules/mocha/bin/mocha "test/unit/**/*.js"`, but I'm not sure if it's worth it (as we'll be running this script through npm anyway)

Reply:
I think it is a little improvement so putting this into my changelist